pkg/readychecker/types/built_in.go

```diff
@@ -62,6 +77,49 @@ func isDeploymentReady(obj runtime.Object) (isReady, retriableError bool, e erro
 		deployment.Status.UpdatedReplicas == replicas, false, nil
 }
 
+func isHorizontalPodAutoscalerReady(obj runtime.Object) (isReady, retriableError bool, e error) {
+	var hpa autoscaling_v2b1.HorizontalPodAutoscaler
+	if err := util.ConvertType(autoscalingV2B1, obj, &hpa); err != nil {
+		return false, false, err
+	}
+
+	// For the HPA to be ready, AbleToScale and ScalingActive conditions should exist and be true
+	// If ScalingActive is false, it could still be coming up *or* it could have failed (e.g. because it can't access metrics)
+	// Can't distinguish this based on the reason, so using a timeout
+	foundAbleToScale := false
+	foundScalingActive := false
+	for _, cond := range hpa.Status.Conditions {
+		switch cond.Type {
+		case hpaAbleToScaleCondition:
+			switch cond.Status {
+			case core_v1.ConditionTrue:
+				foundAbleToScale = true
+			case core_v1.ConditionFalse:
+				// AbleToScale should not be false if the HPA is working, this is a failure
+				return false, false, errors.Errorf("%s: %s", cond.Reason, cond.Message)
+			case core_v1.ConditionUnknown:
+				// Assume it's still coming up
+			}
+		case hpaScalingActiveCondition:
+			switch cond.Status {
+			case core_v1.ConditionTrue:
+				foundScalingActive = true
+			case core_v1.ConditionFalse:
+				// It's expected that it will take a little while to get metrics
+				// If it's been stuck in this condition for >3min we assume it's failed
+				now := meta_v1.Now()
+				if cond.LastTransitionTime.Add(3 * time.Minute).Before(now.Time) {
+					return false, false, errors.Errorf("%s: %s", cond.Reason, cond.Message)
```

Comment:
This also seems like a recoverable error?
e.g. if the autoscale controller was down and now is up again, it will be able to scale.

Reply:
Have made this retriable.